forgerock-auth/src/main/java/org/forgerock/android/auth/centralize/BrowserLauncher.kt

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2024 ForgeRock. All rights reserved.
+ *
+ *  This software may be modified and distributed under the terms
+ *  of the MIT license. See the LICENSE file for details.
+ */
+
+package org.forgerock.android.auth.centralize
+
+import android.content.Intent
+import androidx.activity.ComponentActivity
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.launch
+import net.openid.appauth.AuthorizationResponse
+import org.forgerock.android.auth.AuthorizeContract
+import org.forgerock.android.auth.FRListener
+import org.forgerock.android.auth.FRUser
+import org.forgerock.android.auth.Listener
+import java.lang.IllegalStateException
+
+/**
+ * Singleton class to launch the Browser
+ * Centralize login browser launcher.
+ */
+internal object BrowserLauncher {
+
+    private var launcher: Launcher? = null
+
+    /**
+     * Initialize the Launcher
+     */
+    @Synchronized
+    internal fun init(launcher: Launcher) {
+        BrowserLauncher.launcher = launcher
+    }
+
+    /**
+     * reset the Launcher state
+     */
+    @Synchronized
+    internal fun reset() {
```

Comment:
who is using this ? cant find the caller

Reply:
this call androidx.activity.result.ActivityResultLauncher#unregister
```
Unregisters this launcher, releasing the underlying result callback, and any references captured within it. You should call this if the registry may live longer than the callback registered for this launcher.
```
I did not implement this in our Sample App, because our App has only one Activity, but I can add it back to the Activity.

Reply:
Sample app updated.